### PR TITLE
[SPARK-48961][PYTHON] Make the parameter naming of `PySparkException` consistent with JVM

### PR DESCRIPTION
**Pull request by itholic**
### What changes were proposed in this pull request?

This PR proposes to make the parameter naming of `PySparkException` consistent with JVM

### Why are the changes needed?

The parameter names of `PySparkException` are different from `SparkException` so there is an inconsistency when searching those parameters from error logs.

SparkException:
https://github.com/apache/spark/blob/6508b1f5e18731359354af0a7bcc1382bc4f356b/common/utils/src/main/scala/org/apache/spark/SparkException.scala#L27-L33

PySparkException:
https://github.com/apache/spark/blob/6508b1f5e18731359354af0a7bcc1382bc4f356b/python/pyspark/errors/exceptions/base.py#L29-L40

### Does this PR introduce _any_ user-facing change?

The error parameter names are changed from:
- `error_class` -> `errorClass`
- `message_parameters` -> `messageParameters`
- `query_contexts` -> `context`

### How was this patch tested?

The existing CI should pass

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

**Comment by itholic** [comment]
cc @HyukjinKwon

File: python/docs/source/development/logger.rst [hunk 1]
```diff
@@ -52,7 +52,7 @@ Example log entry:
         "file": "/path/to/file.py",
         "line_no": "17",
```
Comment (HyukjinKwon) [inline python/docs/source/development/logger.rst:53]: What about line_no?

Comment (itholic) [reply python/docs/source/development/logger.rst:53]: This indicates the line number retrieved from stack frame by using `frame.lineno`.
On second thought maybe we should use `lineno` or `line_number` instead `line_no`?

Comment (HyukjinKwon) [reply python/docs/source/development/logger.rst:53]: yeah

Comment (itholic) [reply python/docs/source/development/logger.rst:53]: Sounds good! Let me submit a quick followup for https://github.com/apache/spark/pull/47145 as I think this is technically not related to this PR.

Comment (itholic) [reply python/docs/source/development/logger.rst:53]: Created followup here: https://github.com/apache/spark/pull/47437

**Review by HyukjinKwon — APPROVED** [review]
LGTM if tests pass

**Comment by itholic** [comment]
Merged into master. Thanks!

**Pull request closed without merging**